### DISCUSSION TITLE
Suggestion for the best way for an iOS developer to learn to use `composeModifier`?

### DISCUSSION THREAD

**Discussion by ptrkstr** [skiptools/skip#213] [Q&A]
Hi all, huge fan of Skip Tools!

Apologies for the slightly generic title but I wanted to demonstrate my perspective as an iOS developer (with limited Android knowledge) trying to use `composeModifier`:

1. Place a `TextEditor`.
2. Require this `TextEditor` to take up the full height.
3. Look at the Skip [source code](https://github.com/skiptools/skip-ui/blob/55cebefa82bc050e3dc04502a758cf7394d2e249/Sources/SkipUI/SkipUI/Text/TextEditor.swift#L39) for `TextEditor` and see there is `modifier: context.modifier.fillWidth()`
4. Sheepishly assume that there is a modifier called `fillHeight()`
5. Copy [all of these imports](https://github.com/skiptools/skip-ui/blob/55cebefa82bc050e3dc04502a758cf7394d2e249/Sources/SkipUI/SkipUI/Text/TextEditor.swift#L7-L21) from the docs (cause one of them is probably for a `fillHeight()` right? 👀
6. Add to the TextEditor: `composeModifier { $0.fillHeight() }`
7. Experience error `@Composable invocations can only happen from the context of a @Composable function`
8. Try `$0.fillWidth()` just to double check that would work
9. Same error

Could you suggest how an iOS developer would know which import to use (I'm guessing there is no auto complete) and how to best use `composeModifier`?

**Comment by aabewhite** [comment]
My first question would be: why do you need to use composeModifier in the first place? Did you try the "SwiftUI way" first, and what went wrong?

ComposeView and composeModifier are generally for when you know you need a certain Android-specific solution

**Reply by ptrkstr** [reply]
Thanks you for the quick reply @aabewhite 🙏 

> Did you try the "SwiftUI way" first, and what went wrong?

Good question, my use case was wanting the Android TextEditor to take up the full height of the parent which is what the iOS TextEditor does I believe. Perhaps this is a change needed in Skip to make Android behave like iOS? You can see the difference in the screenshot below.

<img height="400" alt="image" src="https://github.com/user-attachments/assets/5a4bf1a3-7196-4bbb-8f05-422b2af64c87">

**Reply by aabewhite** [reply]
Thanks! I'll look into this. In the meantime, you can in fact use .composeModifier, but not with some of Skip's internal modifiers. The following should work:

```
TextEditor(...)
    #if SKIP
    .composeModifier { $0.fillMaxHeight() }
    #endif
```

**Reply by ptrkstr** [reply]
Perfect thank you!
Feel free to close this whenever you'd like.

**Reply by aabewhite** [reply]
OK, TextEditor layout and styling is fixed in skip-ui 1.6.1. Use File->Packages->Update to Latest Versions to get it

**Reply by ptrkstr** [reply]
Thanks for the quick turnaround @aabewhite, can confirm iOS and Android `TextEditor`s match 👌
